lib/galaxy/model/database_heartbeat.py

```diff
@@ -73,26 +79,17 @@ def is_config_watcher(self, value):
         for callback in self._observers:
             callback(self._is_config_watcher)
 
-    @property
-    def worker_process(self):
-        return (
-            self.sa_session.query(WorkerProcess)
-            .with_for_update(of=WorkerProcess)
-            .filter_by(
-                server_name=self.server_name,
-                hostname=self.hostname,
-            )
-            .first()
-        )
-
     def update_watcher_designation(self):
-        worker_process = self.worker_process
-        if not worker_process:
-            worker_process = WorkerProcess(server_name=self.server_name, hostname=self.hostname)
-        worker_process.update_time = now()
-        worker_process.pid = self.pid
-        self.sa_session.add(worker_process)
-        self.sa_session.flush()
+        expression = self._worker_process_identifying_clause()
+        stmt = select(WorkerProcess).with_for_update(of=WorkerProcess).where(expression)
+        with self.sa_session() as session:
+            with session.begin():
+                worker_process = session.scalars(stmt).first()
+                if not worker_process:
+                    worker_process = WorkerProcess(server_name=self.server_name, hostname=self.hostname)
+                worker_process.update_time = now()
+                worker_process.pid = self.pid
+                session.add(worker_process)
```

Comment:
```suggestion
        with self.sa_session() as session, session.begin():
            worker_process = session.scalars(stmt).first()
            if not worker_process:
                worker_process = WorkerProcess(server_name=self.server_name, hostname=self.hostname)
            worker_process.update_time = now()
            worker_process.pid = self.pid
            session.add(worker_process)
```
super minor, but less nesting is always nice IMO.

Reply:
Agreed!